pkg/virt-controller/watch/vm.go

```diff
@@ -2644,14 +2637,23 @@ func (c *VMController) sync(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachin
 			syncErr = &syncErrorImpl{fmt.Errorf("Error encountered while handling CPU change request: %v", err), HotPlugCPUErrorReason}
 		}
 
+		var vmUpdateErr error
 		if syncErr == nil {
 			if !equality.Semantic.DeepEqual(vm, vmCopy) {
-				vm, err = c.clientset.VirtualMachine(vmCopy.Namespace).Update(context.Background(), vmCopy)
-				if err != nil {
-					syncErr = &syncErrorImpl{fmt.Errorf("Error encountered when trying to update vm according to add volume and/or memory dump requests: %v", err), FailedUpdateErrorReason}
+				vm, vmUpdateErr = c.clientset.VirtualMachine(vmCopy.Namespace).Update(context.Background(), vmCopy)
+				if vmUpdateErr != nil {
+					syncErr = &syncErrorImpl{fmt.Errorf("Error encountered when trying to update vm according to add volume and/or memory dump requests: %v", vmUpdateErr), FailedUpdateErrorReason}
 				}
 			}
 		}
+
+		if vmUpdateErr == nil && c.clusterConfig.HotplugNetworkInterfacesEnabled() {
+			updatedVMIfaces := vmispec.IndexInterfaceSpecByName(vm.Spec.Template.Spec.Domain.Devices.Interfaces)
+			if err = c.applyDynamicIfaceRequestOnVMI(vmi, vmCopy.Status.InterfaceRequests, updatedVMIfaces); err != nil {
+				log.Log.Object(vm).Errorf("error encountered when trying to apply interface request on vmi: %v", err)
```

Comment:
This is not needed, for the same reason the `Update` does not have it.
The `syncErr` will eventually be reported.